### PR TITLE
@orta=>presentation mode settings vc

### PR DESCRIPTION
**Pull request by sarahscott**
- [x] skeleton
- [x] functionality
- [x] tests

simulator screen shot nov 3 2015 5 48 56 pm

### CODE REVIEW

File: Classes/Controllers/Settings/ARLabSettingsPresentationModeViewController.m [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    if ([partner hasConfidentialNotes]) {
+        [presentationModeOptions addObject:@{
+            AROptionsKey : ARHideArtworkEditButton,
+            AROptionsName : @"Hide Artwork Edit Button"
+        }];
+    }
+
+    self.presentationModeOptions = [NSArray arrayWithArray:presentationModeOptions];
```
Comment (orta) [inline Classes/Controllers/Settings/ARLabSettingsPresentationModeViewController.m]: :+1:

Comment (orta) [inline Classes/Controllers/Settings/ARLabSettingsPresentationModeViewController.m]: though ideally, you want to use `_presentationModeOptions =` as another subclass can change the `setPresentationModeOptions:` function

File: Classes/Controllers/Settings/ARLabSettingsPresentationModeViewController.m [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    NSString *option = self.presentationModeOptions[indexPath.row][AROptionsKey];
+
+    BOOL on = ![self.defaults boolForKey:option];
+    [self.defaults setBool:on forKey:option];
+    [self.defaults synchronize];
+
+    ARToggleSwitch *toggle = (id)[tableView cellForRowAtIndexPath:indexPath].accessoryView;
+    toggle.on = on;
+
+    [[NSNotificationCenter defaultCenter] postNotificationName:ARUserDidChangeGridFilteringSettingsNotification object:nil];
```
Comment (orta) [inline Classes/Controllers/Settings/ARLabSettingsPresentationModeViewController.m]: `self.defaults`

**Comment by sarahscott** [comment]
to be finished after the settings structure pr is merged

**Comment by sarahscott** [comment]
a few things in this PR are noted as being temporary - i will make an issue for them so i remember to change them when we get rid of the old settings.

i'm also making a new issue for Hide Not For Sale Works & Hide Unpublished Works functionalities so this PR doesn't get too massive.

File: ArtsyFolio Tests/View Controllers/ARLabSettingsPresentationModeViewControllerTests.m [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        publishedArtwork.availability = @"sold";
+        
+        Artwork *unpublishedArtwork = genericArtworkInContext(context);
+        unpublishedArtwork.isPublished = @(NO);
+        unpublishedArtwork.isAvailableForSale = @(NO);
+        unpublishedArtwork.confidentialNotes = @"fjdkalfjdks";
+        
+        subject.defaults = (id)onDefaults();
+        
+        expect(subject).to.haveValidSnapshot();
```
Comment (orta) [inline ArtsyFolio Tests/View Controllers/ARLabSettingsPresentationModeViewControllerTests.m]: this isn't showing all as true in the snapshot, this known?

File: ArtsyFolio Tests/View Controllers/ARLabSettingsPresentationModeViewControllerTests.m [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+});
+
+describe(@"setting defaults", ^{
+    it(@"sets the right default when tapped", ^{
+        Artwork *artwork = genericArtworkInContext(context);
+        artwork.displayPrice = @"2";
+        
+        subject.defaults = (id)offDefaults();
+        
+        NSIndexPath *path = [NSIndexPath indexPathForRow:0 inSection:0];
```
Comment (orta) [inline ArtsyFolio Tests/View Controllers/ARLabSettingsPresentationModeViewControllerTests.m]: :+1:

File: ArtsyFolio Tests/View Controllers/ARLabSettingsPresentationModeViewControllerTests.m [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    return [ForgeriesUserDefaults defaults:@{
+        ARHideUnpublishedWorks : @(YES),
+        ARShowPrices : @(YES),
+        ARShowConfidentialNotes : @(YES),
+        ARHideWorksNotForSale : @(YES),
+        ARHidePricesForSoldWorks : @(YES)
+    }];
+}
+
+NSInteger numberOfRowsIn(ARLabSettingsPresentationModeViewController *subject)
```
Comment (orta) [inline ArtsyFolio Tests/View Controllers/ARLabSettingsPresentationModeViewControllerTests.m]: cool

File: Classes/Categories/Partner+InventoryHelpers.h [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+/// Does the partner have any works with prices?
+- (BOOL)hasWorksWithPrice;
+
+/// Does the partner have any sold works with prices?
+- (BOOL)hasSoldWorksWithPrices;
+
+/// Does the partner have any confidential notes associated with their artworks?
+- (BOOL)hasConfidentialNotes;
+
+@end
```
Comment (orta) [inline Classes/Categories/Partner+InventoryHelpers.h]: :+1:

**Comment by orta** [comment]
:+1: Alright, this looks good

**Pull request merged**